libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -74,30 +76,34 @@ import com.mapbox.navigation.ui.maps.camera.view.MapboxRouteOverviewButton
 import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
 import com.mapbox.navigation.ui.speedlimit.view.MapboxSpeedLimitView
 import com.mapbox.navigation.ui.tripprogress.view.MapboxTripProgressView
+import com.mapbox.navigation.ui.utils.internal.lifecycle.ViewLifecycleRegistry
+import com.mapbox.navigation.ui.utils.internal.lifecycle.keepExecutingWhenStarted
 import com.mapbox.navigation.ui.voice.view.MapboxSoundButton
 import com.mapbox.navigation.utils.internal.ifNonNull
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.flowOf
-import kotlinx.coroutines.launch
 import java.lang.ref.WeakReference
 import java.util.concurrent.CopyOnWriteArraySet
 
 @OptIn(ExperimentalCoroutinesApi::class)
-class NavigationView : ConstraintLayout {
+class NavigationView : ConstraintLayout, LifecycleOwner {
 
     val navigationViewApi: MapboxNavigationViewApi by lazy {
         MapboxNavigationViewApiImpl(this)
     }
-    private val lifeCycleOwner: LifecycleOwner
-    private val activity: FragmentActivity by lazy {
-        try {
-            context as FragmentActivity
-        } catch (exception: ClassCastException) {
-            throw ClassCastException(
-                "Please ensure that the provided Context is a valid FragmentActivity"
-            )
-        }
+    private val viewLifecycleRegistry: ViewLifecycleRegistry by lazy {
+        val hostingLifecycleOwner = context.unwrapIfNeeded() as? LifecycleOwner
+            ?: throw LifeCycleOwnerNotFoundException()
+        ViewLifecycleRegistry(
+            hostingLifecycleOwner,
+            localLifecycleOwner = this
+        )
+    }
```

Comment:
Think we should make `ViewLifecycleRegistry` take the responsibility for making this clean inside the `NavigationView`.

```suggestion
    private val viewLifecycleRegistry = ViewLifecycleRegistry(this)
```

Then we can iterate on a clean interface for `ViewLifecycleRegistry`

``` kotlin
@UiThread
class ViewLifecycleRegistry(view: View) : LifecycleRegistry(checkOwner(view)) {

    private var isAttached = false
    private val hostingLifecycleOwner: LifecycleOwner by lazy {
        view.context.unwrapIfNeeded() as LifecycleOwner
    }
...
private fun checkOwner(view: View): LifecycleOwner {
    val lifecycleOwner = view as? LifecycleOwner
    checkNotNull(lifecycleOwner) {
        "Your view must implement LifecycleOwner"
    }
    return lifecycleOwner
}

private fun Context.unwrapIfNeeded(): Context {
    return if (this !is Activity && this is ContextWrapper) {
        this.baseContext.unwrapIfNeeded()
    } else {
        this
    }
}

```

Reply:
Looks good, adding here :+1:

Reply:
nit: make `checkOwner` and `unwrapIfNeeded` extension as static (companion object or separate file): they don't need state of the class